scripts/gas-refund-program/staking/safety-module-stakes-tracker.ts

```diff
@@ -374,8 +389,56 @@ export default class SafetyModuleStakesTracker
     this.differentialStates.bptPoolTotalSupply.sort(timeseriesComparator);
   }
 
-  compute_BPT_to_PSP_Rate(timestamp: number) {
-    const pspBalance = reduceTimeSeries(
+  async resolveBPTBalanceOfStkPSPBptChanges() {
+    const events = (
+      await Promise.all([
+        bptAsERC20.queryFilter(
```

Comment:
Can this cooler run into 10000 (don't remember exact number) event limit?

Reply:
Several parts uses query filtering. As the script runs continuously the interval is at most 1 epoch long (2w).

It’s only a problem for the hedge case where you run the script from the first epoch though we’re still far from this limit. I thought it wasn’t worth tackling yet. But surely should be done in the future.

Reply:
Édit: it can happen if during an epoch we get a storm of events, sorry